makewhatis traverses manpages twice, as default manpath contains two
directories that are symlinked together ... this isn't a real fix, just
a workaround that should be OK for the way Gentoo does things.  a real
fix would be to filter out any directories which resolv to the same
directory ...

--- man-1.5o2/src/makewhatis.sh
+++ man-1.5o2/src/makewhatis.sh
@@ -41,7 +41,7 @@
 # and should be first.
 # It is a bug to add /var/cache/man to DEFCATPATH.
 dm=
-for d in /usr/man /usr/share/man /usr/X11R6/man /usr/local/man
+for d in /usr/local/man /usr/share/man
 do
     if [ -d $d ]; then
 	if [ x$dm = x ]; then dm=$d; else dm=$dm:$d; fi
@@ -49,7 +49,7 @@
 done
 DEFMANPATH=$dm
 dc=
-for d in /usr/man/preformat /usr/man /usr/share/man/preformat /usr/share/man
+for d in /usr/local/man /usr/share/man
 do
     if [ -d $d ]; then
 	if [ x$dc = x ]; then dc=$d; else dc=$dc:$d; fi
